mlflow/genai/evaluation/utils.py

```diff
@@ -24,13 +24,57 @@
         EvaluationDatasetTypes = Union[pd.DataFrame, list[dict], EvaluationDataset]
 
 
-# TODO: ML-52299
 def _convert_to_legacy_eval_set(data: "EvaluationDatasetTypes") -> dict:
     """
-    Takes in different types of inputs and converts it into to the current eval-set schema that
-    Agent Evaluation takes in. The transformed schema should be accepted by mlflow.evaluate()
+    Takes in a dataset in the format that mlflow.genai.evaluate() expects and converts it into
+    to the current eval-set schema that Agent Evaluation takes in. The transformed schema should
+    be accepted by mlflow.evaluate().
+    The expected schema can be found at:
+    https://docs.databricks.com/aws/en/generative-ai/agent-evaluation/evaluation-schema
     """
-    raise NotImplementedError("This function is not implemented yet.")
+    column_mapping = {
+        "inputs": "request",
+        "outputs": "response",
+        "trace": "trace",
+    }
+
+    if isinstance(data, list):
+        # validate that every item in the list is a dict and has inputs as key
+        for item in data:
+            if not isinstance(item, dict):
+                raise ValueError("Every item in the list must be a dictionary.")
```

Comment:
```suggestion
                raise MlflowException.invalid_parameter_value("Every item in the list must be a dictionary.")
```

nit: We prefer using `MlflowException` in general.